### PR TITLE
Handle exception if no default DNS search domain found

### PR DESCRIPTION
**Pull request by JamesGibo**

### CODE REVIEW

File: nmoscommon/mdns/dnsUtils.py [hunk 1]
```diff
@@ -15,12 +15,16 @@
...
 
 
 def _defaultDomain():
-    resolve = dns.resolver.Resolver()
-    return str(resolve.search[0])
+    try:
+        resolve = dns.resolver.Resolver()
+        return str(resolve.search[0])
+    except IndexError: 
+        print("No default search domain found in /etc/resolv.conf")
```
Comment (andrewbonney) [inline nmoscommon/mdns/dnsUtils.py]: Could this be removed? The log statement when the exception is caught should be sufficient.

**Comment by andrewbonney** [comment]
Resolves #82

File: nmoscommon/mdns/dnsServiceController.py [hunk 2]
```diff
@@ -17,7 +17,7 @@
 from nmoscommon.mdns.dnsListener import DNSListener
 from nmoscommon.mdns.dnsService import DNSService
 from nmoscommon.mdns import dnsUtils
-from nmoscommon.mdns.mdnsExceptions import DNSRecordNotFound
+from nmoscommon.mdns.mdnsExceptions import DNSRecordNotFound, NoDefaultDnsSearchDomian
```
Comment (andrewbonney) [inline nmoscommon/mdns/dnsServiceController.py]: Just spotted a minor thing on a second pass, 'Domian' should be 'Domain' in the exception naming. This seems to be consistent everywhere so will work, but should probably be fixed.

**Comment by JamesGibo** [comment]
Required changes made

**Pull request merged**